datalad/distributed/create_sibling_ria.py

```diff
@@ -0,0 +1,586 @@
+    reduce the number of inodes consumed (no checkout + potential archive) as
+    well as it allows to resolve dependencies (that is (sub)datasets) merely by
+    their id.  Finally, there is a file "ria-layout-version" put beneath the
+    store's base path, determining the version of the dataset tree layout and a
+    file of the same name per each dataset directory determining object tree
+    layout version (we already switch from dirhashlower to dirhashmixed for
+    example) and an additional directory "error_logs" at the toplevel.  """
+
+    # TODO: description?
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to process.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        url=Parameter(
+            args=("url",),
+            metavar="ria+<ssh|file>://<host>[/path]",
+            doc="""URL identifying the target RIA store and access protocol.
+            """,
+            constraints=EnsureStr() | EnsureNone()),
+        name=Parameter(
+            args=('-s', '--name',),
+            metavar='NAME',
+            doc="""Name of the sibling.
+            With `recursive`, the same name will be used to label all
+            the subdatasets' siblings.""",
+            constraints=EnsureStr() | EnsureNone(),
+            required=True),
+        ria_remote_name=Parameter(
+            args=("--ria-remote-name",),
+            metavar="NAME",
+            doc="""Name of the RIA remote (a git-annex special remote).
+            Must not be identical to the sibling name. If not specified,
+            defaults to the sibling name plus a '-ria' suffix.""",
+            constraints=EnsureStr() | EnsureNone()),
+        post_update_hook=Parameter(
+            args=("--post-update-hook",),
+            doc="""Enable git's default post-update-hook for the created
+            sibling.""",
+            action="store_true"),
+        shared=Parameter(
+            args=("--shared",),
+            metavar='{false|true|umask|group|all|world|everybody|0xxx}',
+            doc="""If given, configures the permissions in the
+            RIA store for multi-users access.
+            Possible values for this option are identical to those of
+            `git init --shared` and are described in its documentation.""",
+            constraints=EnsureStr() | EnsureBool() | EnsureNone()),
+        group=Parameter(
+            args=("--group",),
+            metavar="GROUP",
+            doc="""Filesystem group for the repository. Specifying the group is
+            crucial when [CMD: --shared=group CMD][PY: shared="group" PY]""",
+            constraints=EnsureStr() | EnsureNone()),
+        ria_remote=Parameter(
+            args=("--no-ria-remote",),
```

Comment:
This name is confusing given the have of the command. May be make it `--no-ria-archive` if that what it is?

Reply:
Potential confusion - true. But it's not about an archive either. It's about whether or not to have a special remote, since it can be useful to have that ID based tree of bare repos without any annex object.

Reply:
I agree with @bpoldrack